Fenrir/Packing.fs

```diff
@@ -0,0 +1,108 @@
+﻿module Fenrir.Packing
+
+open System
+open System.IO
+open Fenrir.Zlib
+
+let getPackPath (gitPath: String) (packFile: String) (extension: String) : String =
+    Path.Combine(gitPath, "objects", "pack", packFile + extension)
+
+let flagToBits(key: String) : byte[] =
+    match key with
+        | "commit" -> [| 0uy; 0uy; 1uy |]
+        | "tree"   -> [| 0uy; 1uy; 0uy |]
+        | "blob"   -> [| 0uy; 1uy; 1uy |]
+        | "tag"    -> [| 1uy; 0uy; 0uy |]
+        | _        -> failwithf "Unknown object type provided"
+
+let anotherEndian (reader: BinaryReader) : int =
+    reader.ReadBytes 4
+        |> Array.map (fun x -> int x)
+        |> Array.reduce (fun x y ->
+            x * 256 + y)
+
+let readHash (reader: BinaryReader) : String =
+    reader.ReadBytes 20
+        |> Array.fold (fun acc elem ->
+            String.Concat(acc, sprintf "%02x" elem)) ""
+
+let byteToBits (num: byte) : byte[] =
+    let mutable number = num
+    Array.init 8 (fun _ ->
+        let result = number % 2uy
+        number <- number >>> 1
+        result)
+
+
+let bitsToInt (bits: byte[]) =
+    let mutable st = 1
+
+    Array.foldBack (fun i acc ->
+        let result = acc + int i * st
+        st <- st * 2
+        result) bits 0
```

Comment:
Honestly, I don't like all this business with bits represented as byte arrays. If you have time, could you please use `BitArray` instead of `bits: byte[]`?

It is not a mandatory change though.

Reply:
Actually you know what? This PR is already pretty big, and you've put much work into it. Let's consider this refactoring as a separate task. I'll make a note to raise that task after merge of this PR.